app/services/widgets/settings/alert-box/alert-box-api.ts

```diff
@@ -0,0 +1,571 @@
+  pledge_show_animation: string;
+  pledge_sound_href: string;
+  pledge_sound_volume: number;
+  pledge_text_animation: string;
+  pledge_text_delay: number;
+  pledge_variations: IAlertBoxVariation[];
+}
+
+interface IAlertBoxTiltifySettings {
+  tiltifydonation_alert_duration: number;
+  tiltifydonation_alert_message_min_amount: number;
+  tiltifydonation_custom_css: string;
+  tiltifydonation_custom_html: string;
+  tiltifydonation_custom_html_enabled: boolean;
+  tiltifydonation_custom_js: string;
+  tiltifydonation_custom_json: string;
+  tiltifydonation_enabled: boolean;
+  tiltifydonation_font: string;
+  tiltifydonation_font_color: string;
+  tiltifydonation_font_color2: string;
+  tiltifydonation_font_size: string;
+  tiltifydonation_font_weight: string;
+  tiltifydonation_hide_animation: string;
+  tiltifydonation_image_href: string;
+  tiltifydonation_layout: string;
+  tiltifydonation_message_font: string;
+  tiltifydonation_message_font_color: string;
+  tiltifydonation_message_font_size: string;
+  tiltifydonation_message_font_weight: string;
+  tiltifydonation_message_template: string;
+  tiltifydonation_show_animation: string;
+  tiltifydonation_sound_href: string;
+  tiltifydonation_sound_volume: number;
+  tiltifydonation_text_animation: string;
+  tiltifydonation_text_delay: number;
+  tiltifydonation_variations: IAlertBoxVariation[];
+}
+
+interface IAlertBoxTreatSettings {
+  treat_alert_duration: number;
+  treat_custom_css: number;
+  treat_custom_html: string;
+  treat_custom_html_enabled: boolean;
+  treat_custom_js: string;
+  treat_custom_json: string;
+  treat_enabled: boolean;
+  treat_font: string;
+  treat_font_color: string;
+  treat_font_color2: string;
+  treat_font_size: string;
+  treat_font_weight: string;
+  treat_hide_animation: string;
+  treat_image_href: string;
+  treat_layout: string;
+  treat_message_font: string;
+  treat_message_font_color: string;
+  treat_message_font_size: string;
+  treat_message_font_weight: string;
+  treat_message_template: string;
+  treat_show_animation: string;
+  treat_sound_href: string;
+  treat_sound_volume: number;
+  treat_text_animation: string;
+  treat_text_delay: number;
+  treat_variations: IAlertBoxVariation[];
+}
+
+export interface IAlertBoxApiSettings extends
+IAlertBoxGeneralSettings,
+IAlertBoxSubSettings,
+IAlertBoxBitsSettings,
+IAlertBoxDonationSettings,
+IAlertBoxFollowSettings,
+IAlertBoxHostSettings,
+IAlertBoxRaidSettings,
+IAlertBoxMerchSettings,
+IAlertBoxResubSettings,
+IAlertBoxTreatSettings,
+IAlertBoxDonorDriveSettings,
+IAlertBoxExtraLifeSettings,
+IAlertBoxGamewsipSettings,
+IAlertBoxJustGivingSettings,
+IAlertBoxPatreonSettings,
+IAlertBoxTiltifySettings {}
+
+// SLOBS GENERAL SETTINGS
+export interface IAlertBoxSetting {
+  enabled: boolean;
+  showMessage: boolean;
+  showResubMessage?: boolean;
+  variations: IAlertBoxVariation[]
+}
+
+export interface IAlertBoxSettings extends IAlertBoxGeneralSettings {
```

Comment:
This is a beastly file.  Props for defining an interface around this whole thing.